src/main/java/com/uber/cadence/internal/replay/HistoryHelper.java

```diff
@@ -196,12 +200,17 @@ public DecisionEvents next() {
       List<HistoryEvent> decisionEvents = new ArrayList<>();
       List<HistoryEvent> newEvents = new ArrayList<>();
       boolean replay = true;
-      long replayCurrentTimeMilliseconds = -1;
       long nextDecisionEventId = -1;
       while (events.hasNext()) {
         HistoryEvent event = events.next();
         EventType eventType = event.getEventType();
-        // hasNext is for queries that do not have DecisionTaskStarted at the end of the history.
+
+        // Sticky workers receive an event history that starts with DecisionTaskCompleted
+        if (eventType == EventType.DecisionTaskCompleted) {
```

Comment:
Shouldn't the condition be 
`eventType == EventType.DecisionTaskCompleted && nextDecisionEventId == -1`
?

Reply:
Yes it should. Well, later completed events always follow started events so it gets handled in the code below. But it's safer/cleaner to add that condition so i'll add it.